clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/AvroGenericDaVinciClient.java

```diff
@@ -795,12 +806,28 @@ public synchronized void start() {
       this.keyDeserializer = FastSerializerDeserializerFactory.getFastAvroGenericDeserializer(keySchema, keySchema);
       this.genericRecordStoreDeserializerCache =
           new AvroStoreDeserializerCache(daVinciBackend.get().getSchemaRepository(), getStoreName(), true);
-      this.storeDeserializerCache = clientConfig.isSpecificClient()
-          ? new AvroSpecificStoreDeserializerCache<>(
+
+      if (clientConfig.isSpecificClient()) {
+        if (daVinciConfig.isRecordTransformerEnabled()) {
+          if (recordTransformerConfig.getOutputValueClass() != clientConfig.getSpecificValueClass()) {
```

Comment:
Let's do a test case through these, I think looking back at the integration test cases the transformed type is always the same.

Reply:
I have a couple of [unit tests](https://github.com/kvargha/venice/blob/83a88bd89be67696ca39ae513689941fbe265ea8/clients/da-vinci-client/src/test/java/com/linkedin/davinci/client/AvroGenericDaVinciClientTest.java#L109-L128) that go over this. Unless you mean changing the types? Ex: Int -> String?

Reply:
Yeah, let's do a type change